pkg/types/vsphere/validation/platform.go

```diff
@@ -41,5 +41,8 @@ func ValidatePlatform(p *vsphere.Platform, fldPath *field.Path) field.ErrorList
 		}
 	}
 
+	if len(p.Cluster) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("cluster"), "must specify the cluster"))
```

Comment:
If this is required for IPI, but not for UPI, then we can leave it optional here (but maybe log a warning?).  And then die around [here][1] if the value is unset?  Also, I think @abhinavdahiya raised some good concerns with these new properties [back][2] [here][3].  Working out the consuming code when you add new properties would make it easier to get these sorts of wrinkles sorted earlier in development.

[1]: https://github.com/openshift/installer/blob/75ad924123661d417a3257bfbdc68beae5811959/pkg/asset/cluster/tfvars.go#L433
[2]: https://github.com/openshift/installer/pull/2875#discussion_r363070297
[3]: https://github.com/openshift/installer/pull/2875#issuecomment-570864578